python/tvm/micro/project.py

```diff
@@ -92,17 +92,16 @@ class TemplateProject:
     """Defines a glue interface to interact with a template project through the API Server."""
 
     @classmethod
-    def from_directory(cls, template_project_dir, options):
-        return cls(client.instantiate_from_dir(template_project_dir), options)
+    def from_directory(cls, template_project_dir):
+        return cls(client.instantiate_from_dir(template_project_dir))
 
-    def __init__(self, api_client, options):
+    def __init__(self, api_client):
         self._api_client = api_client
-        self._options = options
         self._info = self._api_client.server_info_query(__version__)
         if not self._info["is_template"]:
             raise NotATemplateProjectError()
 
-    def generate_project(self, graph_executor_factory, project_dir):
+    def generate_project(self, graph_executor_factory, project_dir, options):
```

Comment:
just noting for posterity that originally i didn't do this because i wanted all calls to Project API server to use the same options, but since there is just one call used with TemplateProject, and you might actually want to examine `self._info` before constructing `options`, this makes more sense to me

Reply:
I see your point and that was exactly what I thought when I was going for that change. Thanks your comment, it really helps me to understand the design better and increase my confidence I'm following the motivations behind correctly :)